src/main/java/com/google/devtools/build/lib/server/GrpcServerImpl.java

```diff
@@ -595,6 +597,20 @@ private void executeCommand(RunRequest request, BlockingStreamObserver<RunRespon
             .map(arg -> arg.toString(StandardCharsets.ISO_8859_1))
             .collect(ImmutableList.toImmutableList());
 
+        if (args.isEmpty() && !request.getParameterFile().isEmpty()) {
+          ImmutableList.Builder<String> builder = ImmutableList.builder();
+          File file = new File(request.getParameterFile());
+          try {
+            Scanner sc = new Scanner(file);
+            sc.useDelimiter("\u0000");
+            while (sc.hasNext()) {
+              builder.add(sc.next());
+            }
+            args = builder.build();
+          } catch (IOException e) {
+          }
+        }
```

Comment:
Do you really need to modify the Java code?
I think it would be simpler to open the file and do the expansion in `option_processor.cc`.
(also skip the change to the proto file)

Reply:
Yes, after doing the expansion in the cpp client we still need to send it to the java server. As the server/client communicate over gRPC this still has a message size limit, see https://github.com/bazelbuild/bazel/issues/8609#issuecomment-571830880 for some more context.

The only solution I found for that is making sure the client writes the arguments to a file and instructs the server to read that file, for which we'd need a proto change (or implicitly encode this somehow in the existing field for args).

Reply:
I'm told that the gRPC limit is entirely artificial and we can just increase it.

Call .maxInboundMessageSize(1<<30) on NettyServerBuilder here: 
https://github.com/bazelbuild/bazel/blob/636430a30f3eef86062640d66e049df37ad16711/src/main/java/com/google/devtools/build/lib/server/GrpcServerImpl.java#L439

Reply:
Let's keep a sensible limit on the max inbound message size.

Reply:
@michajlo what would you consider a sensible limit? 10/100s of megabytes? 

What I like about the file approach is that it doesn't impose another seemingly arbitrarily chosen limit. What I'd like to avoid is ending up with a choice that needs to be increased 6months or a year down the line. Maybe the limit should be configurable?